src/main/java/com/iexec/sms/tee/session/TeeSessionService.java

```diff
@@ -78,7 +83,13 @@ public String generateTeeSession(
                 .generateSecureSession(sessionYmlAsString.getBytes())
                 .getStatusCode()
                 .is2xxSuccessful();
-        return isSessionGenerated ? sessionId : "";
+        if (!isSessionGenerated) {
+            throw new TeeSessionGenerationException(
+                    SECURE_SESSION_GENERATION_FAILED,
```

Comment:
```suggestion
                    SECURE_SESSION_POST_FAILED,
```
?

Reply:
What about `SECURE_SESSION_CAS_FAILED`? I find both of our propositions too specific. But maybe users don't want to bother with CAS?

Reply:
I like the idea of adding CAS explicitly yes, there is just a verb missing in your proposition